management-api-agent-common/src/main/java/com/datastax/mgmtapi/NodeOpsProvider.java

```diff
@@ -737,98 +740,106 @@ public void clearSnapshots(
   @Rpc(name = "repair")
   public String repair(
       @RpcParam(name = "keyspaceName") String keyspace,
-      @RpcParam(name = "tables") List<String> tables,
+      @RpcParam(name = "tables") Optional<List<String>> tables,
       @RpcParam(name = "full") Boolean full,
-      @RpcParam(name = "notifications") boolean notifications)
+      @RpcParam(name = "notifications") boolean notifications,
+      @RpcParam(name = "repairParallelism") Optional<RepairParallelism> repairParallelism,
```

Comment:
> The second issue is more of a confusing thing. The v2 model object looks a lot like the Cassandra object, but they are not the same. And in the Resource code, we are not sending the Cassandra object, we are sending the V2 model object. So this will never deserialize correctly since we are expecting the Cassandra object in this method.

Woops... Sorry, I should have spotted that one :( 

Wouldn't we be better off adding a generic serializer for the collections interface to make things more generic? I have to be honest, I do not like the way management API is not using the type system very much throughout. I understand some of that is unavoidable due to the way the RPC calls work, but if we can't even serialise our own types, that is unfortunate.

Reply:
Actually, I need to understand why we aren't just serializing straight to json or using some off the shelf serialization here - was there a reason for writing our own serializer?